### PR TITLE
project_panel: Select the newly created file when copy/pasting a file

### LINKED ISSUE #14361: Select the newly created file when copy/pasting a file
### Check for existing issues

- [X] Completed

### Describe the feature

When a file is copied and pasted, the old file remains selected, but I think we all are used to have the newly created file selected so we can rename it or do whatever we want with it.

### If applicable, add mockups / screenshots to help present your vision of the feature

_No response_

### PR DESCRIPTION
**Pull request by CharlesChen0823**
close: #14361 

Release Notes:

- N/A

### CODE REVIEW

**Comment by CharlesChen0823** [comment]
Don't known if this modify is needed. the CI failed because selected target had changed which is expected.

**Review by SomeoneToIgnore — COMMENTED** [review]
> the CI failed because selected target had changed which is expected.

If it's expected, we have to update the tests then?

I am also not sure about this test failure:
https://github.com/zed-industries/zed/actions/runs/9984980450/job/27595098698?pr=14705#step:4:919

seems that we've copied `a` in that test, selected `root` and pasted it there twice:
https://github.com/zed-industries/zed/blob/eb7fe5745363b7b0f9fb1a5bd863c7a5ac5af3c2/crates/project_panel/src/project_panel.rs#L3676-L3680

with one `a` already existing under `root`.

I would expect that there are 2 new entries, `root/a copy` and `root/a copy 1` and either of them is selected.
I would prefer the latter, but the code is written to prefer the former.
Yet, the test failure shows, that `root/a copy/a/` is selected?? Is that the desired behavior?
Maybe I misread the test, and then we just have to adjust the test results, but please double check that part.

File: crates/project_panel/src/project_panel.rs [hunk 1]
```diff
@@ -1318,15 +1320,36 @@ impl ProjectPanel {
...
-                        .detach_and_log_err(cx)
+                    let task = self.project.update(cx, |project, cx| {
+                        project.copy_entry(clipboard_entry.entry_id, new_path, cx)
+                    });
+                    tasks.push(task);
                 }
             }
+
+            cx.spawn(|this, mut cx| async move {
+                let mut entry_ids = Vec::new();
```
[the remark this answers is not in the document]
Comment (CharlesChen0823) [reply crates/project_panel/src/project_panel.rs]: done

File: crates/project_panel/src/project_panel.rs [hunk 2]
```diff
@@ -1318,15 +1320,36 @@ impl ProjectPanel {
...
+                for task in tasks {
+                    if let Some(entry) = task.await.log_err() {
+                        if let Some(entry) = entry {
+                            entry_ids.push(entry.id);
+                        }
+                    }
+                }
+                if let Some(id) = entry_ids.first() {
+                    this.update(&mut cx, |this, _cx| {
```
[the remark this answers is not in the document]
Comment (CharlesChen0823) [reply crates/project_panel/src/project_panel.rs]: done

**Comment by CharlesChen0823** [comment]
> Yet, the test failure shows, that root/a copy/a/ is selected?? Is that the desired behavior?
I think this is desired behavior, and I also check this behavior in VSCode, same in.

**Review by SomeoneToIgnore — APPROVED** [review]
Thank you for confirming that now things work as you'd expect.

**Pull request merged**